corehq/util/workbook_reading/adapters/xlsx.py

```diff
@@ -59,13 +60,27 @@ def _make_cell_value(self, cell):
                 return cell.value
         return cell.value
 
-    def iter_rows(self):
+    @property
+    @memoized
+    def _max_row(self):
+        # self._worksheet.max_row is the max_row with data OR WITH FORMATTING
+        # That means that if formatting is applied, that will be the xlsx row limit
+        # Note that this is 1-indexed for consistency with openpyxl
+        max_row = 1
         for row in self._worksheet.iter_rows():
+            if any(cell.value for cell in row):
+                max_row = row[0].row
+        return max_row
+
+    def iter_rows(self):
+        for i, row in enumerate(self._worksheet.iter_rows(), 1):
+            if i > self._max_row:
```

Comment:
If ```self._max_row``` is 1-indexed, should this be ```>=```?

Reply:
I did `enumerate(..., 1)` so they'd be consistent